yt_dlp/YoutubeDL.py

```diff
@@ -4031,6 +4049,17 @@ def _opener(self):
         handler = self._request_director.handlers['Urllib']
         return handler._get_instance(cookiejar=self.cookiejar, proxies=self.proxies)
 
+    def get_impersonate_targets(self):
+        return sorted(self._request_director.collect_from_handlers(
+            lambda rh: [(*target, rh.RH_NAME) for target in rh.get_supported_targets()],
+            [lambda _, v: isinstance(v, ImpersonateRequestHandler)]
```

Comment:
Cant we use `_check_extensions({'impersonate': ()})` instead of `isinstance`?

Reply:
an impersonate request handler doesn't have to support the impersonate extension

we could do both, but I think it's unnecessary

Reply:
> an impersonate request handler doesn't have to support the impersonate extension

How so?

Reply:
the request handlers are designed so the extension is not required. it's not a reliable way to test.

There is a seperate base class ImpersonateRequestHandler for this reason